closed/src/java.base/share/classes/sun/security/ec/NativeECDHKeyAgreement.java

```diff
@@ -150,13 +179,15 @@ protected Key engineDoPhase(Key key, boolean lastPhase)
             throw new IllegalStateException
                 ("Only two party agreement supported, lastPhase must be true");
         }
-        if (!(key instanceof PublicKey)) {
+        if (!(key instanceof ECPublicKey)) {
```

Comment:
This can capture the downcast for use on line 188:
```java
        if (!(key instanceof ECPublicKey ecKey)) {
```

Reply:
Changed.